bridges/otellogr/logsink.go

```diff
@@ -4,10 +4,58 @@
 // Package otellogr provides a [LogSink], a [logr.LogSink] implementation that
 // can be used to bridge between the [logr] API and [OpenTelemetry].
 //
+// # Record Conversion
+//
+// The logr records are converted to OpenTelemetry [log.Record] in the following
+// way:
+//
+//   - Time is set as the current time of conversion.
+//   - Message is set as the Body using a [log.StringValue].
+//   - Level is transformed and set as the Severity. The SeverityText is not
+//     set.
+//   - KeyAndValues are transformed and set as Attributes.
+//   - Name is logged as an additional attribute with the key "logger".
+//   - The [context.Context] value in KeyAndValues is propagated to OpenTelemetry
+//     log record. All non-nested [context.Context] values are ignored and not
+//     added as attributes. If there are multiple [context.Context] the last one
+//     is used.
+//
+// The Level is transformed by using the [WithLevels] option. If the option is
+// not provided, all logr levels will be mapped to [log.SeverityInfo].
+// The levels are mapped in order, with the first level in the list corresponding
+// to the logr level 0, the second level corresponding to logr level 1, and so on.
+// If the logr level is greater than the number of levels provided, it will be
+// mapped to [log.SeverityInfo].
```

Comment:
Trying out a different level transformation

Original attempt https://github.com/scorpionknifes/opentelemetry-go-contrib/blob/beae842225fd1a14f5a935cd4e419b0e519e3ac9/bridges/otellogr/example_test.go#L23-L33 
Open to suggestions